### DISCUSSION TITLE
Plans for OpenSSL 3.x Support in Fluent Bit?

### DISCUSSION THREAD

**Discussion by Sathya-Manalan** [fluent/fluent-bit#9569] [Q&A] [answered]
Hello Fluent Bit Community,

I’m reaching out to inquire about Fluent Bit's plans for supporting OpenSSL 3.x. Currently, Fluent Bit seems to rely on OpenSSL 1.1.x, as seen with the dependency on libssl.so.1.1, which is causing issues for users running systems with OpenSSL 3.x (e.g., Debian/Ubuntu 22.04 and later, or RHEL 9, which have migrated to OpenSSL 3.x).

Encountered the following error when OpenSSL 3 was set as the default.

> `fluent-bit: error while loading shared libraries: libssl.so.1.1: cannot open shared object file: No such file or directory`

Given that OpenSSL 3.x was released with significant changes, including updated library names and improved security features, I wanted to know if there are any plans to update Fluent Bit’s dependencies to OpenSSL 3.x. Specifically:

Is there an existing roadmap or planned release for OpenSSL 3.x support?
Will Fluent Bit eventually move to OpenSSL 3.x as the default, or is this support going to remain optional (with some kind of backward compatibility)?
If OpenSSL 3.x is adopted, are there any major changes users should be aware of that may require configuration adjustments?
I believe this change would help users who are running more recent distributions with OpenSSL 3.x and improve compatibility with modern environments.

Looking forward to hearing your thoughts!

Thanks,
Sathya Manalan

**Comment by patrick-stephens** [comment, marked as answer]
There is an open issue on this, we have migrated to OpenSSL 3 for more recent targets: https://github.com/fluent/fluent-bit/issues/7644

Contribution is welcome for those affecting you.
